use-cases.md

```diff
@@ -69,11 +65,7 @@ The `operation` above refers to the name the server wants to give to the Span. F
 The `span` object above can be `None` if the Tracer did not find relevant headers in the incoming request: presumably because the client did not send them. In this case the server needs to start a brand new trace.
 
 ```python
-    span = tracer.join_trace_from_text(
-        operation_name=operation,
-        context_snapshot=request.headers,
-        trace_attributes=request.headers
-    )
+    span = tracer.get_span_extractor(request.class).extract(request)
```

Comment:
(starting a different thread here since I don't want to muddy the discussion above with @yurishkuro)

@michaelsembwever I think this is an appealing direction and I hope we can do something along these lines. Of course nothing is ever so simple, though. :)  I do have a few concerns...

1. There is a "stuttering" aspect to these code snippets: namely that the type of an object is used to get an Injector/Extractor, then that object is subsequently passed in to the inject/extract method. One obvious alternative would be to consolidate around a single method for the common case... basically a `Tracer.extract(obj, extractor=None)` sort of signature. If `extractor` is unspecified, then `extract` makes its best guess based on `obj` and goes from there. I also prefer this approach because it doesn't force use into guessing the correct extractor based only on the type information: for instance, there might be more than one way to encode data into a `string->string` map.
1. Related to that last point, I like how this design allows the decision to keep/drop trace attributes a per-Injector/Extractor concern. So, again, with the sort of signature I've proposed above, we could have something like `Tracer.extract(request, extractor=StringMapExtractor(attrs=False))`.
1. Of course something would be wrong with the world if I didn't quibble about naming... and I would like to do so if we can come to agreement on the higher-order bits above.